### PR TITLE
Fix shellcheck errors in scripts (part 2)

### LINKED ISSUE #915: [TASK] Perform shellcheck on shell scripts
Shell scripts are risky, but the `shellcheck` tool can help to find issues with it.
- [x] (merged) Delete deprecated scripts
- [x] (**all on review**) Fix existing shellcheck issues
          (number of shellcheck issues)
   - [x] ( 1) ./ci/owncloud/upload_to_owncloud.sh
   - [x] ( 1) ./ci/run_easymesh_cert.sh
   - [x] ( 2) ./docker-builder.sh
   - [x] ( 2) ./docker-runner.sh
   - [x] ( 4) ./framework/platform/bpl/uci/scripts/platform_daemon.sh
   - [x] ( 1) ./tests/openwrt/test_status.sh
   - [x] ( 2) ./tests/test_gw_repeater.sh
   - [x] ( 6) ./tools/docker/builder/openwrt/build.sh
   - [x] ( 1) ./tools/docker/builder/rdkb/toolchain/bitbake.sh
   - [x] ( 6) ./tools/docker/build.sh
   - [x] ( 1) ./tools/docker/clang-format.sh
   - [x] (18) ./tools/docker/image-build.sh
   - [x] (10) ./tools/docker/image-pull.sh
   - [x] ( 1) ./tools/docker/runner/entrypoint.sh
   - [x] ( 3) ./tools/docker/run.sh
   - [x] ( 4) ./tools/docker/static-analysis/cppcheck.sh
   - [x] ( 2) ./tools/docker/stop.sh
   - [x] ( 7) ./tools/docker/test.sh
   - [x] ( 3) ./tools/functions.sh
   - [x] ( 2) ./tools/klocwork/kw.sh

- [x] (Canceled) Add shellcheck to the builder docker image
- [x] (**on review**) Run shellcheck in gitlab CI.

## Total lists
**Note:** Lists below will be updated based on discussion
Removal candidates:
-`./build/common/beerocks/scripts/prplmesh_utils.sh`
-`./build/install/scripts/prplmesh_utils.sh`
-`./build/out/scripts/prplmesh_utils.sh`
-`./common/beerocks/scripts/dhcp.sh`
-`./docker-builder.sh`
-`./docker-runner.sh`
-`./tests/openwrt/test_status.sh`
-`./tools/maptools.sh`
-`./tools/scripts/prplMesh-platform-init.sh`
-`./tools/commands/deploy_rdkb.sh`
-`./tools/commands/deploy_ugw.sh`
-`./framework/platform/bpl/uci/scripts/platform_daemon.sh`

No action due to soon removal:
-`./tests/test_flows.sh`
-`./tests/test_gw_repeater.sh`

Scripts not mentioned in any of lists above will be fixed.

### PR DESCRIPTION
**Pull request by MuteSpirit**
Fixing shellcheck issues in 2nd part of scripts mentioned in task #915

Description of fixed shellcheck issues you may found in commit comments

### CODE REVIEW

File: tools/docker/builder/openwrt/build.sh [hunk 1]
```diff
@@ -82,7 +85,7 @@ main() {
     while true; do
         case "$1" in
             -h | --help)               usage; exit 0; shift ;;
-            -b | --build-options)      BUILD_OPTIONS="$2"; shift; shift ;;
+            -b | --build-options)      BUILD_OPTIONS+=($(echo -n "$2")); shift; shift ;;
```
Comment (rmelotte) [inline tools/docker/builder/openwrt/build.sh]: Here I think the safest is to drop '--build-options' entirely. I think nobody use it anymore.

```suggestion
```

File: docker-builder.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/bin/sh
 
-scriptdir="$(cd "${0%/*}"; pwd)"
-${scriptdir}/tools/docker/build.sh "$@"
+scriptdir="$(cd "${0%/*}" || exit 1; pwd)"
```
Comment (rmelotte) [inline docker-builder.sh:3]: Like for the other PR, I think `$(cd "${0%/*}" && pwd)` is less confusing (exit might make you think it exits the script)
(nitpick)

File: tools/docker/builder/openwrt/build.sh [hunk 3]
```diff
@@ -1,12 +1,13 @@
-#!/bin/sh -e
+#!/bin/bash -e
 
 scriptdir="$(cd "${0%/*}"; pwd)"
 rootdir="${scriptdir%/*/*/*/*}"
 
+# shellcheck source=../../../../tools/functions.sh
```
Comment (rmelotte) [inline tools/docker/builder/openwrt/build.sh]:
```suggestion
# shellcheck source=../../../functions.sh
```

File: tools/docker/image-build.sh [hunk 4]
```diff
@@ -6,27 +6,29 @@
 # See LICENSE file for more details.
 ###############################################################
 
-scriptdir="$(cd "${0%/*}"; pwd)"
+scriptdir="$(cd "${0%/*}" || exit 1; pwd)"
 rootdir="${scriptdir%/*/*}"
 
-. ${rootdir}/tools/functions.sh
+# shellcheck source=../../tools/functions.sh
```
Comment (rmelotte) [inline tools/docker/image-build.sh:12]:
```suggestion
# shellcheck source=../functions.sh
```

**Review by rmelotte — APPROVED** [review]
Only nitpicks, LGTM :+1:

File: tools/docker/builder/openwrt/build.sh (hunk 1 above)
Comment (MuteSpirit) [reply tools/docker/builder/openwrt/build.sh]: Ok, i'll do that if almost one more reviewer will approve your suggestion

File: docker-runner.sh [hunk 5]
```diff
@@ -1,4 +1,4 @@
 #!/bin/sh
 
-scriptdir="$(cd "${0%/*}"; pwd)"
-${scriptdir}/tools/docker/run.sh "$@"
+scriptdir="$(cd "${0%/*}" || exit 1; pwd)"
```
Comment (arnout) [inline docker-runner.sh:3]: As mentioned elsewhere: I prefer
```suggestion
scriptdir="$(cd "${0%/*}" && pwd)"
```
but that's really personal preference.

File: tools/docker/builder/openwrt/build.sh (hunk 3 above)
Comment (arnout) [reply tools/docker/builder/openwrt/build.sh]: I disagree - with `../../../../tools/functions.sh` it corresponds one-to-one with the expansion of `$rootdir/tools/functions.sh`

File: tools/docker/builder/openwrt/build.sh [hunk 6]
```diff
@@ -1,12 +1,13 @@
-#!/bin/sh -e
+#!/bin/bash -e
```
Comment (arnout) [inline tools/docker/builder/openwrt/build.sh]: Please mention something like this very explicitly in the commit message. Not just that we changed the shell to bash, but also why.

File: tools/docker/builder/openwrt/build.sh (hunk 1 above)
Comment (arnout) [reply tools/docker/builder/openwrt/build.sh]: I agree!

File: tools/docker/builder/openwrt/build.sh [hunk 7]
```diff
@@ -72,7 +72,7 @@ main() {
         exit 1
     fi
 
-    if ! OPTS=$(getopt -o 'hb:d:io:r:t:v' --long help,build-options:,device:,image,openwrt-version:,openwrt-repository:,tag:,verbose -n 'parse-options' -- "$@"); then
+    if ! OPTS=$(getopt -o 'hb:d:io:r:t:v' --long help,build-options:,target-device:,image,openwrt-version:,openwrt-repository:,tag:,verbose -n 'parse-options' -- "$@"); then
```
Comment (arnout) [inline tools/docker/builder/openwrt/build.sh]: Nice catch!

File: tools/docker/build.sh [hunk 8]
```diff
@@ -1,37 +1,38 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (arnout) [inline tools/docker/build.sh:1]: Again, please explain why in commit message.

File: tools/docker/build.sh [hunk 9]
```diff
@@ -1,37 +1,38 @@
...
 ###############################################################
 # SPDX-License-Identifier: BSD-2-Clause-Patent
 # SPDX-FileCopyrightText: 2019-2020 the prplMesh contributors (see AUTHORS.md)
 # This code is subject to the terms of the BSD+Patent license.
 # See LICENSE file for more details.
 ###############################################################
 
-scriptdir="$(cd "${0%/*}"; pwd)"
+scriptdir="$(cd "${0%/*}" || exit 1; pwd)"
```
Comment (arnout) [inline tools/docker/build.sh:9]: Again, slight preference for `&&`

File: tools/docker/build.sh [hunk 10]
```diff
@@ -1,37 +1,38 @@
...
-    --workdir=$rootdir --user=${SUDO_UID:-$(id -u)}:${SUDO_GID:-$(id -g)} \
-    -e USER=${SUDO_USER:-${USER}} -v ${rootdir}:${rootdir} \
-    --entrypoint=./tools/maptools.py \
-    "
+    docker_args=(
+        --workdir "${rootdir}"
+        --user "${SUDO_UID:-$(id -u)}:${SUDO_GID:-$(id -g)}"
+        -e "USER=${SUDO_USER:-${USER}}"
+        -v "${rootdir}:${rootdir}"
+        --entrypoint "./tools/maptools.py"
```
Comment (arnout) [inline tools/docker/build.sh:27]: I really like how it's much easier to read now!

File: tools/docker/build.sh [hunk 11]
```diff
@@ -1,37 +1,38 @@
...
+        run "${scriptdir}"/image-build.sh
     }
 
     # Default docker arguments
-    docker_args="\
-    --workdir=$rootdir --user=${SUDO_UID:-$(id -u)}:${SUDO_GID:-$(id -g)} \
-    -e USER=${SUDO_USER:-${USER}} -v ${rootdir}:${rootdir} \
-    --entrypoint=./tools/maptools.py \
-    "
+    docker_args=(
```
Comment (arnout) [inline tools/docker/build.sh:22]: (nitpick) Now we're using bash, we can also 
```suggestion
    local docker_args=(
```

File: tools/docker/image-build.sh [hunk 12]
```diff
@@ -19,12 +19,11 @@ usage() {
     echo "      -h|--help - show this help menu"
     echo "      -v|--verbose - verbosity on"
     echo "      -b|--base-image - Base OS image to use (Dockerfile 'FROM')"
-    echo "      -n|--native - Use the same base OS image as the running system"
```
Comment (arnout) [inline tools/docker/image-build.sh:22]: As mentioned in the other PR: Please include an explanation in the commit message why this didn't work.

File: tools/docker/image-build.sh [hunk 13]
```diff
@@ -6,13 +6,14 @@
 # See LICENSE file for more details.
 ###############################################################
 
-scriptdir="$(cd "${0%/*}"; pwd)"
+scriptdir="$(cd "${0%/*}" || exit 1; pwd)"
```
Comment (arnout) [inline tools/docker/image-build.sh:9]:
```suggestion
scriptdir="$(cd "${0%/*}" && pwd)"
```

File: tools/docker/image-build.sh (hunk 4 above)
Comment (arnout) [reply tools/docker/image-build.sh:12]: I disagree

**Review by arnout — APPROVED** [review]
Many nitpicks, nothing major.

File: tools/docker/builder/openwrt/build.sh (hunk 1 above)
Comment (abelog) [reply tools/docker/builder/openwrt/build.sh]: Than drop it is. This option is removed.

**Pull request merged**